Move quick actions inside of flexbox with injectable component set

colinking opened the pull request:
It is difficult to extend the quick action list, since the current quick actions are inserted at once inside of the ThreadListQuickActions class. I've moved these into an injectable component set, which makes it possible to inject new quick actions that can be re-ordered to the front via a flexbox.

Here's an example:
![image](https://cloud.githubusercontent.com/assets/2907397/12084817/5dac18cc-b286-11e5-8aef-17b049325295.png)

mbilker commented:
Please remove trailing white-space that is causing the Travis build to throw linting errors. Also, the [component register commands](https://github.com/colinking/N1/blob/d46ba0d5d5d9a3aa7f95b83091eb1c1495af5b6d/internal_packages/thread-list/lib/thread-list.cjsx#L142-L145) should not be placed inside the resolver method. It will register multiple copies of the components.

mbilker commented:
Looks good otherwise!

colinking commented:
Thanks for the feedback, I've fixed the linting issues. I looked into InjectedComponentSets and moved the default quick actions under the `children` attribute which should be a cleaner solution as compared to using the ComponentRegistry.

bengotow commented:
Hey @colinking — this looks great! Definitely a nice feature. Let me do a bit of profiling and make sure this won't impact the performance of the thread list too much and we can land it. By default, InjectedComponentSet wraps each child in an "UnsafeComponent", which calls `React.render` separately for it's contents. That means that a bad third-party component can't break anything outside it, but causes the React render cycle to be fragmented a bit more. Should be fine, but we can double-check.

bengotow commented:
Looks good! Since you moved the existing items to `children`, there's actually no performance penalty unless there are plugins adding items.
